app/src/main/java/com/limelight/ShortcutTrampoline.java

```diff
@@ -0,0 +1,253 @@
+                                                    getResources().getString(R.string.conn_error_title),
+                                                    getResources().getString(R.string.scut_not_paired),
+                                                    true);
+                                        }
+
+                                        // We don't want any more callbacks from now on, so go ahead
+                                        // and unbind from the service
+                                        if (managerBinder != null) {
+                                            managerBinder.stopPolling();
+                                            unbindService(serviceConnection);
+                                            managerBinder = null;
+                                        }
+                                    }
+                                });
+                            }
+                        }
+                    });
+                }
+            }.start();
+        }
+
+        public void onServiceDisconnected(ComponentName className) {
+            managerBinder = null;
+        }
+    };
+
+    protected boolean validateInput() {
+        //Validate UUID
+        try {
+            UUID.fromString(uuidString);
+        } catch (IllegalArgumentException ex) {
+            Dialog.displayDialog(ShortcutTrampoline.this,
+                    getResources().getString(R.string.conn_error_title),
+                    getResources().getString(R.string.scut_invalid_uuid),
+                    true);
+            return false;
+        }
+
+        //Validate App ID (if provided)
+        if(appIdString != null && appIdString.isEmpty()) {
+            try {
+                Integer.parseInt(appIdString);
+            } catch (NumberFormatException ex) {
+                Dialog.displayDialog(ShortcutTrampoline.this,
+                        getResources().getString(R.string.conn_error_title),
+                        getResources().getString(R.string.scut_invalid_app_id),
+                        true);
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        UiHelper.notifyNewRootView(this);
+
+        uuidString = getIntent().getStringExtra(AppView.UUID_EXTRA);
+        appIdString = getIntent().getStringExtra(APP_ID_EXTRA);
+
+        if(validateInput()) {
+            // Bind to the computer manager service
+            bindService(new Intent(this, ComputerManagerService.class), serviceConnection,
+                    Service.BIND_AUTO_CREATE);
+
+            blockingLoadSpinner = SpinnerDialog.displayDialog(this, getResources().getString(R.string.conn_establishing_title),
+                    getResources().getString(R.string.applist_connect_msg), true);
+        } else {
+            finish();
```

Comment:
If you finish the activity here, it will close your error dialogs too (likely without the user being able to see them). You can probably just skip this, return, and let the `Dialog.displayDialog()` handle finishing this activity after the dialog is closed.

Reply:
This is probably the bug you mention at the end of your PR summary